genesyscloud/architect_grammar/data_source_genesyscloud_architect_grammar_test.go

```diff
@@ -49,3 +53,31 @@ func generateGrammarDataSource(
 	}
 	`, resourceID, name, dependsOnResource)
 }
+
+func cleanupArchitectGrammar(idPrefix string) {
+	architectApi := platformclientv2.NewArchitectApi()
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+		architectGrammars, _, getErr := architectApi.GetArchitectGrammars(pageNum, pageSize, "", "", nil, idPrefix, "", "", false)
+		if getErr != nil {
+			log.Printf("failed to get architect grammar %s", getErr)
+			return
+		}
+
+		if architectGrammars.Entities == nil || len(*architectGrammars.Entities) == 0 {
+			break
+		}
+
+		for _, grammar := range *architectGrammars.Entities {
+			if grammar.Name != nil {
+				_, _, delErr := architectApi.DeleteArchitectGrammar(*grammar.Id)
+				if delErr != nil {
+					log.Printf("failed to delete architect grammar %s", delErr)
+					return
```

Comment:
should this be continue here ?

Reply:
we are deleting the resources containing the prefix of the resource name, so if its not deleted properly ( rare case ) , the testcase sometimes fails. 
These cleanup functions were used long before. I re-used the same logic here - did not do any new changes to the clean up functions.